src/main/scala/net/ripe/rpki/publicationserver/PublicationMessages.scala

```diff
@@ -71,15 +83,15 @@ case class ReplyMsg(pdus: Seq[ReplyPdu]) extends Msg {
     val sb = new StringBuilder
     pdus.foreach { pdu =>
       val textual = pdu match {
-        case PublishR(uri, Some(tag)) => s"""<publish tag="$tag" uri="$uri"/>"""
-        case PublishR(uri, None) => s"""<publish uri="$uri"/>"""
-        case WithdrawR(uri, Some(tag)) => s"""<withdraw tag="$tag" uri="$uri"/>"""
-        case WithdrawR(uri, None) => s"""<withdraw uri="$uri"/>"""
-        case ListR(uri, hash, Some(tag)) => s"""<list tag="$tag" uri="$uri" hash="$hash"/>"""
-        case ListR(uri, hash, None) => s"""<list uri="$uri" hash="$hash"/>"""
+        case PublishR(uri, Some(tag)) => s"""<publish tag="${attr(tag)}" uri="${uri.toASCIIString}"/>"""
```

Comment:
nit/being pedantic: Not sure this is enough to make sure the string is URI safe, for example `">` would mess up the XML and  is in ascii.

This may be as safe as not doing toASCIIString.

Reply:
Indeed. Added it there.